iptables/features_detect_test.go

```diff
@@ -102,12 +102,42 @@ func TestFeatureDetection(t *testing.T) {
 				MASQFullyRandom:     false,
 			},
 		},
+		{
+			"iptables v1.8.2 (legacy)",
+			"error",
+			Features{
+				RestoreSupportsLock: true,
+				SNATFullyRandom:     false,
+				MASQFullyRandom:     false,
+			},
+		},
+		{
+			"iptables v1.8.2 (nf_tables)",
+			"error",
+			Features{
+				RestoreSupportsLock:          true,
+				SNATFullyRandom:              false,
+				MASQFullyRandom:              false,
+				IptablesRestoreHasReplaceBug: true,
+			},
+		},
+		{
+			"iptables v1.8.2 (legacy)",
+			"error",
+			Features{
+				RestoreSupportsLock:          true,
+				SNATFullyRandom:              false,
+				MASQFullyRandom:              false,
+				IptablesRestoreHasReplaceBug: false,
+			},
+		},
 	} {
 		tst := tst
 		t.Run("iptables version "+tst.iptablesVersion+" kernel "+tst.kernelVersion, func(t *testing.T) {
 			RegisterTestingT(t)
 			dataplane := newMockDataplane("filter", map[string][]string{}, "legacy")
-			featureDetector := NewFeatureDetector()
+			featureDetector := NewFeatureDetector("nft")
```

Comment:
Hmm, it's not really right, if backend mode is passed as "nft", to have cases above saying "(legacy)", is it?  I understand the test intention, to cover all the cases as easily as possible, but I wonder if Felix should panic if it saw that for real.  Perhaps it would be better to partition the cases and run them with either `NewFeatureDetector()` or `NewFeatureDetector("nft")` as appropriate.  WDYT?

Reply:
Agree, it's got a bit messy